flexget/components/sites/sites/hebits.py

```diff
@@ -0,0 +1,262 @@
+
+
+class SearchHeBits:
+    """
+        HEBits search plugin.
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'user_name': {'type': 'string'},
+            'password': {'type': 'string'},
+            'category': {'type': 'string', 'enum': [_.name for _ in HEBitsCategory]},
+            'free': {'type': 'boolean'},
+            'double': {'type': 'boolean'},
+            'triple': {'type': 'boolean'},
+            'pack': {'type': 'boolean'},
+            'order_by': {
+                'type': 'string',
+                'enum': [_.name for _ in HEBitsSort],
+                'default': 'date',
+            },
+            'order_desc': {'type': 'boolean', 'default': True},
+        },
+        'required': ['user_name', 'password'],
+        'additionalProperties': False,
+    }
+
+    base_url = 'https://hebits.net/'
+    login_url = f"{base_url}takeloginAjax.php"
+    search_url = f"{base_url}browse.php"
+    download_url = f"{base_url}downloadRss.php"
+    profile_link = f"{base_url}my.php"
+
+    @staticmethod
+    def _extract_passkey(user_profile_html: HTML) -> str:
+        """Extracts the passkey from the user profile"""
+        logger.debug('trying to extract passkey')
+        for tr in user_profile_html.find("tr"):
+            td = tr.find("td.pror", first=True)
+            if not td:
+                continue
+            if td.full_text == "פאסקי":
+                logger.debug('succesfully extracted passkey')
+                return tr.find("td.prol", first=True).text
+        else:
+            raise plugin.PluginError(
+                'Could not fetch passkey from user profile, did the page change?'
+            )
+
+    @staticmethod
+    def save_cookies_to_db(user_name: str, cookies: RequestsCookieJar):
+        logger.debug('Saving or updating HEBits cookie in db')
+        expires = datetime.fromtimestamp(glom(cookies, Iter().first()).expires)
+        with Session() as session:
+            cookie = HEBitsCookies(user_name=user_name, cookies=dict(cookies), expires=expires)
+            session.merge(cookie)
+
+    @staticmethod
+    def load_cookies_from_db(user_name: str) -> Optional[RequestsCookieJar]:
+        logger.debug('Trying to load hebits cookies from DB')
+        with Session() as session:
+            saved_cookie = (
+                session.query(HEBitsCookies).filter(HEBitsCookies.user_name == user_name).first()
+            )
+            if saved_cookie and saved_cookie.expires and saved_cookie.expires >= datetime.now():
+                logger.debug('Found valid login cookie')
+                return cookiejar_from_dict(saved_cookie.cookies)
+
+    def login(self, user_name: str, password: str) -> RequestsCookieJar:
+        data = dict(username=user_name, password=password)
+        logger.debug('Trying to login to hebits with user name {}', user_name)
+        rsp = requests.post(self.login_url, data=data)
+        if rsp.text != 'OK':
```

Comment:
Can't we look at the http code?

Reply:
it's always 200. The text is number of attempts left or `OK` for success.